lib/tasks/orchestration.rake

```diff
@@ -0,0 +1,74 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  Orchestration maintainance tasks.
+
+  WARNING: Always backup server data (e.g. DHCP leases file, DNS journal files)
+  prior running these tasks. They do nothing by default, unless perform=1 is
+  specified.
+
+  Examples:
+  rake orchestration:dhcp:add_missing subnet_name=NAME
+    Preview missing DHCP records on a DHCP Smart Proxy.
+
+  rake orchestration:dhcp:add_missing subnet_name=NAME perform=1
+    Do create missing DHCP records on a DHCP Smart Proxy.
+
+  rake orchestration:dhcp:remove_offending subnet_name=NAME
+    Preview offending DHCP records on a DHCP Smart Proxy.
+
+  rake orchestration:dhcp:remove_offending subnet_name=NAME perform=1
+    Do remove offending DHCP records on a DHCP Smart Proxy.
+END_DESC
+
+def fetch_dhcp_records(subnet)
+  subnet.dhcp_proxy.subnet(subnet.network)["reservations"].map do |rec|
+    Net::DHCP::Record.new(rec.merge(proxy: subnet.dhcp_proxy, network: subnet.network))
+  end
+end
+
+namespace :orchestration do
+  namespace :dhcp do
+    task :add_missing => :environment do
+      User.as_anonymous_admin do
+        subnet_name = ENV['subnet_name']
+        dry_run = !ENV['perform']
+        subnet = Subnet.unscoped.find_by_name(subnet_name)
+        raise("Subnet '#{subnet_name}' not found") unless subnet
```

Comment:
You could move these 2 raises into `fetch_dhcp_records` to avoid duplication between the two

Reply:
Yeah that's much cleaner.